p2p/ipld/read.go

```diff
@@ -43,8 +45,17 @@ func RetrieveBlockData(
 			if err != nil {
 				return types.Data{}, err
 			}
-
-			go sc.retrieveShare(rootCid, true, row, col, dag)
+			sem.Acquire(context.Background(), 1)
+			go func(
```

Comment:
I'm new to go so not sure if this is needed but passing in all arguments to resolve a warning https://stackoverflow.com/questions/40326723/go-vet-range-variable-captured-by-func-literal-when-using-go-routine-inside-of-f

Reply:
This is needed to ensure goroutines does not read from vars that gets rewritten on every loop iteration. In this case, you need to pass through: `rootCid`, `row` and `col`. `dag` and `isRow` won't change on loop iterations, so you can pass them directly to `retrieveShare`, and have -2 vars to the goroutine.